### PR TITLE
avro schema ui and validation

### PR DESCRIPTION
**Pull request by Big-Vi**
1 - Added `https://github.com/apache/avro` for front-end validation.
2 - Added node_module `util` to resolve polyfill issue - [Stackoverflow](https://stackoverflow.com/questions/64402821/module-not-found-error-cant-resolve-util-in-webpack)
3 - Added `https://github.com/hamba/avro` for backend schema validation.

In progress #1061

avro

### CODE REVIEW

**Comment by yanivbh1** [comment]
@Avitaltrifsik FYI
That is amazing @Big-Vi ! Thank you.
Before exposing and releasing the feature, we also need to add support in the different SDKs. At least the major ones are GO, JS, Python. LMK if that is something you want to take on as well.
Huge thank you again.

**Comment by Big-Vi** [comment]
@yanivbh1 Yes. I've worked on Go SDK already and sitting in my local computer. If i get some reviews on this pull request, i can do some needed changes and can create pull request on Go SDK.
Yes. I can work on Python & Javascript SDKs as well. I'm not familiar with other languages(.Net, Rust, & Kotlin)  yet.

**Comment by Avitaltrifsik** [comment]
@Big-Vi Hey I would love it if you can join our [Discord channel](https://discord.gg/DfWFT7fzUu) :)

**Comment by Big-Vi** [comment]
Hi, @Avitaltrifsik I've joined the channel already.

**Comment by Avitaltrifsik** [comment]
@Big-Vi Awesome I DMed you :)

File: server/memphis_handlers_schemas.go
Comment (idanasulinmemphis) [inline server/memphis_handlers_schemas.go:126]: not related to here but we have the "GetFilterDetails" endpoint in which you should also return the Avro type

File: go.mod
Comment (idanasulinmemphis) [inline go.mod:38]: any special reason you chose that specific package rather than the Linkedin package for example?

Comment (Big-Vi) [reply go.mod:38]: No specific reasons other than hamba/avro is 2-3 times faster in encoding/decoding than linkedin package.
But i'm happy to swap hamba/avro with linkedin?
https://github.com/hamba/avro#benchmark

File: ui_src/package.json
Comment (Big-Vi) [reply ui_src/package.json:45]: Screenshot 2023-07-24 at 9 36 34 AM

I had similar experience. as soon as i installed util module, polyfill error disappeared. 

But I just pulled the recent code from upstream repo, It seems compiler not throwing any polyfill error locally. I guess something is fixed in upstream repo. I can remove util module i reckon.

File: server/memphis_handlers_schemas.go
Comment (Big-Vi) [reply server/memphis_handlers_schemas.go:126]: Oh. Yep. missed that one.

**Comment by idanasulinmemphis** [comment]
@Big-Vi I see your comments but not the code fixes, any chances you forgot to commit them?

**Comment by Big-Vi** [comment]
@idanasulinmemphis I like to know if you want me to replace "hamba/avro" with linkedin module before i do code fixes.

**Comment by idanasulinmemphis** [comment]
> @idanasulinmemphis I like to know if you want me to replace "hamba/avro" with linkedin module before i do code fixes.

No that is ok to leave it as is

**Comment by Big-Vi** [comment]
> 

Great. I'll let you know once the fixes are done. 

> > @idanasulinmemphis I like to know if you want me to replace "hamba/avro" with linkedin module before i do code fixes.
> 
> No that is ok to leave it as is

File: ui_src/src/domain/schema/components/createSchema/index.js
Comment (Big-Vi) [reply ui_src/src/domain/schema/components/createSchema/index.js:81]: > Add a note saying that at the moment it is only available here to create schemas but they are not enforced yet on the SDK. Also, make sure that in case of an Avro schema attached to a station not causing issues for connected clients or for clients who trying to produce/consume data

I can't find any better way to prevent connected clients error out if avro schema attached to the station. 
Can we update all the SDKs and merge it all together? I'm all ears if you have any suggestion.
I already worked on Go, JS, Python SDKs. I can create pull request today or tomorrow.

Comment (idanasulinmemphis) [reply ui_src/src/domain/schema/components/createSchema/index.js:81]: So sure please open the PRs on the SDKs and we will merge this one only after it won't crash clients

Comment (idanasulinmemphis) [reply ui_src/src/domain/schema/components/createSchema/index.js:81]: After merging the PRs of the SDKs please also update the readme file on this repo, you will find there a sdk features table

Comment (Big-Vi) [reply ui_src/src/domain/schema/components/createSchema/index.js:81]: > So sure please open the PRs on the SDKs and we will merge this one only after it won't crash clients

Sweet as. Just to let you know I'm only familiar with the above mentioned languages. If somebody can work on the remaining SDKs(Java, .Net, and Rust), that would be great.

Comment (Big-Vi) [reply ui_src/src/domain/schema/components/createSchema/index.js:81]: > After merging the PRs of the SDKs please also update the readme file on this repo, you will find there a sdk features table

Yep. Sure.

Comment (idanasulinmemphis) [reply ui_src/src/domain/schema/components/createSchema/index.js:81]: > > So sure please open the PRs on the SDKs and we will merge this one only after it won't crash clients
> 
> Sweet as. Just to let you know I'm only familiar with the above mentioned languages. If somebody can work on the remaining SDKs(Java, .Net, and Rust), that would be great.

Sure would you mind to open issues on those repositories?

Comment (Big-Vi) [reply ui_src/src/domain/schema/components/createSchema/index.js:81]: > 

Yep. I'm on it.

Comment (Big-Vi) [reply ui_src/src/domain/schema/components/createSchema/index.js:81]: > > > So sure please open the PRs on the SDKs and we will merge this one only after it won't crash clients
> > 
> > 
> > Sweet as. Just to let you know I'm only familiar with the above mentioned languages. If somebody can work on the remaining SDKs(Java, .Net, and Rust), that would be great.
> 
> Sure would you mind to open issues on those repositories?

Issues opened:
https://github.com/memphisdev/memphis.java/issues/42
https://github.com/memphisdev/memphis.net/issues/105

For Rust, there's already an open issue. Not sure i need to open another one.
https://github.com/turulix/memphis-rust-community/issues/15

**Comment by idanasulinmemphis** [comment]
Hi @Big-Vi is it already done and awaiting my review?

**Comment by Big-Vi** [comment]
> Hi @Big-Vi is it already done and awaiting my review?

I fixed the code with the fixes you commented on. Good to review once again.

**Review by idanasulinmemphis — APPROVED** [review]
@Big-Vi approved and merged, I'd like to thank you for this amazing contribution, it is much appreciated.

**Review by idanasulinmemphis — APPROVED** [review]

**Comment by idanasulinmemphis** [comment]
@Big-Vi now it's ok, approved, and merged, will be released in the coming 1.1.2 version - next week

**Pull request merged**
